File: .vscode/.python/PySimpleGUI/Projeto_clima_tempo/clima_tempo.py
from PySimpleGUI import PySimpleGUI as sg
import requests

sg.theme('Reddit')

API_KEY = '69ddec2d6c8acd5a481e1c6959e64262'
BASE_URL = 'http://api.openweathermap.org/data/2.5/weather?'

cidades = [
    # Capitais do Mundo
    'Abu Dhabi', 'Accra', 'Addis Ababa', 'Algiers', 'Andorra la Vella', 'Ankara', 'Antananarivo', 'Asunción', 'Athens', 'Baghdad',
    'Baku', 'Bamako', 'Bangkok', 'Banjul', 'Bucharest', 'Budapest', 'Buenos Aires', 'Cairo', 'Canberra', 'Caracas',
    'Castries', 'Chisinau', 'Colombo', 'Copenhagen', 'Dakar', 'Damascus', 'Dhaka', 'Dili', 'Dodoma', 'Doha',
    'Dubai', 'Dushanbe', 'Edinburgh', 'Jakarta', 'Jerusalem', 'Juba', 'Kabul', 'Kampala', 'Kathmandu', 'Khartoum',
    'Kiev', 'Kigali', 'Kingston', 'Kinshasa', 'Kuala Lumpur', 'Kuwait City', 'La Paz', 'Libreville', 'Lima', 'Lisbon',
    'Ljubljana', 'Lome', 'London', 'Luanda', 'Lusaka', 'Madrid', 'Majuro', 'Malabo', 'Male', 'Managua',
    'Manama', 'Mansoura', 'Maputo', 'Marigot', 'Minsk', 'Mogadishu', 'Monaco', 'Monrovia', 'Montevideo', 'Moroni',
    'Moscow', 'Muscat', 'Nairobi', 'Nassau', 'New Delhi', 'Nouakchott', 'Oslo', 'Ottawa', 'Paris', 'Podgorica',
    'Port Moresby', 'Port au Prince', 'Porto-Novo', 'Prague', 'Rabat', 'Reykjavik', 'Riga', 'Rome', 'San Salvador',
    'Sana', 'São Tomé', 'Sarajevo', 'Seoul', 'Singapore', 'Sofia', 'Stockholm', 'Sucre', 'Tallinn', 'Tashkent',
    'Tbilisi', 'Tokyo', 'Tunis', 'Ulaanbaatar', 'Vaduz', 'Valletta', 'Vatican City', 'Vienna', 'Vientiane', 'Vilnius',
    'Warsaw', 'Washington, D.C.', 'Wellington', 'Yamoussoukro', 'Yaoundé', 'Zagreb', 'Zurich',
    # Capitais do Brasil
    'Brasília', 'São Paulo', 'Rio de Janeiro', 'Belo Horizonte', 'Salvador', 'Fortaleza', 'Curitiba', 'Recife', 'Porto Alegre', 'Manaus',
    'Belém', 'Goiânia', 'São Luís', 'Maceió', 'Natal', 'João Pessoa', 'Campo Grande', 'Teresina', 'Aracaju', 'Vitória',
    # Principais Cidades do Brasil
    # Região Norte
    'Boa Vista', 'Macapá', 'Rio Branco', 'Palmas', 'Itacoatiara', 'Santarem', 'Marabá', 'Parintins', 'Humaitá', 'Porto Velho',
    # Região Nordeste
    'Aracaju', 'Campina Grande', 'Feira de Santana', 'Juazeiro', 'Mossoró', 'São Luís', 'Teresina', 'Ilhéus', 'Barreiras', 'Cruz das Almas',
    # Região Centro-Oeste
    'Aparecida de Goiânia', 'Anápolis', 'Dourados', 'Rondonópolis', 'Várzea Grande', 'Chapadão do Sul', 'Luziania', 'Araxá', 'Uberlândia', 'Sorriso',
    # Região Sudeste
    'São José dos Campos', 'Santos', 'Sorocaba', 'Campinas', 'Piracicaba', 'Jundiaí', 'Bauru', 'São Bernardo do Campo', 'São Caetano do Sul', 'Diadema',
    # Região Sul
    'Joinville', 'Blumenau', 'Caxias do Sul', 'Ponta Grossa', 'Maringá', 'Londrina', 'Chapecó', 'Ijuí', 'Santa Maria', 'Pelotas'
]

def obter_clima_tempo(cidade):
    url = BASE_URL + f"q={cidade}&appid={API_KEY}&units=metric&lang=pt_br"
    #Requisita HTTP
    resposta = requests.get(url)
    #Converter em JSON
    dados = resposta.json()

    if resposta.status_code == 200 and dados['cod'] == 200:
        longitude = dados["coord"]["lon"]
        latitude = dados["coord"]["lat"]
        condicao = dados["weather"][0]["main"]
        descricao = dados["weather"][0]["description"]
        temperatura = dados["main"]["temp"]
        sensação_termica = dados["main"]["feels_like"]
        temp_min = dados["main"]["temp_min"]
        temp_max = dados["main"]["temp_max"]
        humidade = dados["main"]["humidity"]
        velocidade_vento = dados["wind"]["speed"]
        pais_sigla = dados["sys"]["country"]
        local_nome = dados["name"]

        return longitude, latitude, condicao, descricao, temperatura, sensação_termica, temp_min, temp_max, humidade, velocidade_vento, pais_sigla, local_nome
    else:
        return None
    
color = 'lightgrey'   
layout = [

    [sg.Text('Escolha uma cidade: ', background_color = color, relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center')], 
    
    [sg.Combo(cidades, key = '-CIDADE-', readonly=True), sg.Button('Buscar', button_color = ('white', 'blue'), font= ('default', 10, 'bold')), sg.Button('Fechar', button_color = ('white', 'blue'), font= ('default', 10, 'bold'))],
  
    [sg.Text('', size = (1,1), background_color = color)],
  
    [sg.Text('Longitude:', size=(15, 1), relief=sg.RELIEF_RIDGE, background_color = color, font=('Default', 10, 'bold'), justification='center'), sg.Text('', key='-LONGITUDE-', background_color = color)],
 
    [sg.Text('Latitude:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-LATITUDE-', background_color = color)],
  
    [sg.Text('Condição:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-CONDICAO-', background_color = color)],
   
    [sg.Text('Descrição:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-DESCRICAO-', background_color = color)],
    
    [sg.Text('', size=(1,1), background_color = color)],
    
    [sg.Text('Temperatura:', size=(15, 1), font=('Default', 14, 'bold'), justification='center', background_color = color, relief=sg.RELIEF_RIDGE,), sg.Text('', key='-TEMPERATURA-', font=('Default', 14), background_color = color)],
    
    [sg.Text('', size=(1,1), background_color = color)],
    
    [sg.Text('Sensação Térmica:', size=(15, 1),relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-SENSACAO_TERM-', background_color = color)],
    
    [sg.Text('Temp Min:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-TEMP_MIN-', background_color = color)],
    
    [sg.Text('Temp Max:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-TEMP_MAX-', background_color = color)],
    
    [sg.Text('Humidade:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-HUMIDADE-', background_color = color)],
    
    [sg.Text('Veloc. do Vento:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-VENTO-', background_color = color)],
    
    [sg.Text('País:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-PAIS-', background_color = color)],
    
    [sg.Text('Local:', size=(15, 1), relief=sg.RELIEF_RIDGE, font=('Default', 10, 'bold'), justification='center', background_color = color), sg.Text('', key='-LOCAL-', background_color = color)]
]

janela = sg.Window("Dados do Clima", layout, size=(320, 500), finalize=True, grab_anywhere=True, background_color = color)
janela.move_to_center()


while True:
    eventos, valores = janela.read()
    if eventos == sg.WIN_CLOSED or eventos == 'Fechar':
        break

    if eventos == "Buscar":
        cidades = valores['-CIDADE-']
        if cidades:
            dados_clima = obter_clima_tempo(cidades)
            if dados_clima is None:
                sg.popup(f'Erro: Dados não encontrados. Tente novamente mais tarde!')
            else:
                janela['-LONGITUDE-'].update(f"{dados_clima[0]}°", font =('Default', 10, 'bold'))
                janela['-LATITUDE-'].update(f"{dados_clima[1]}°", font =('Default', 10, 'bold'))
                janela['-CONDICAO-'].update(dados_clima[2], font =('Default', 10, 'bold'))
                janela['-DESCRICAO-'].update(dados_clima[3], font =('Default', 10, 'bold'))
                janela['-TEMPERATURA-'].update(f"{dados_clima[4]}°C", font =('Default', 14, 'bold'))
                janela['-SENSACAO_TERM-'].update(f"{dados_clima[5]}°C", font =('Default', 10, 'bold'))
                janela['-TEMP_MIN-'].update(f"{dados_clima[6]}°C", font =('Default', 10, 'bold'))
                janela['-TEMP_MAX-'].update(f"{dados_clima[7]}°C", font =('Default', 10, 'bold'))
                janela['-HUMIDADE-'].update(f"{dados_clima[8]}%", font =('Default', 10, 'bold'))
                janela['-VENTO-'].update(f"{dados_clima[9]}m/s", font =('Default', 10, 'bold'))
                janela['-PAIS-'].update(dados_clima[10], font =('Default', 10, 'bold'))
                janela['-LOCAL-'].update(dados_clima[11], font =('Default', 10, 'bold'))               

janela.close()

File: .vscode/.javascript/Clima_Tempo/js/clima_tempo.js
"use strict";

const cidade = document.getElementById("cidade");
const btnBuscar = document.getElementById("btnBuscar");
const btnLimpar = document.getElementById("btnLimpar");
const saida = document.getElementById("saida");
const spanLongitude = document.getElementById("longitude");
const spanLatitude = document.getElementById("latitude");
const spanCandicao = document.getElementById("condicao");
const spanDescricao = document.getElementById("descricao");
const spanTemp = document.getElementById("temp");
const spansensT = document.getElementById("senset");
const spanTempMin = document.getElementById("tempmin");
const spanTempMax = document.getElementById("tempmax");
const spanUmidade = document.getElementById("umidade");
const spanVelVento = document.getElementById("velvento");
const spanPais = document.getElementById("pais");
const spanLocal = document.getElementById("local");

const API_KEY = '0d03d7398ca2f928b965ba5c593360f3'
const BASE_URL = 'http://api.openweathermap.org/data/2.5/weather?';


function apresentarDados(dados)
{
    let longitude = dados["coord"]["lon"]
    let latitude = dados["coord"]["lat"]
    let condicao = dados["weather"][0]["main"]
    let descricao = dados["weather"][0]["description"]
    let temperatura = dados["main"]["temp"]
    let sensação_termica = dados["main"]["feels_like"]
    let temp_min = dados["main"]["temp_min"]
    let temp_max = dados["main"]["temp_max"]
    let umidade = dados["main"]["humidity"]
    let velocidade_vento = dados["wind"]["speed"]
    let pais_sigla = dados["sys"]["country"]
    let local_nome = dados["name"]

    return [longitude, latitude, condicao, descricao, temperatura, sensação_termica, temp_min, temp_max, umidade, velocidade_vento, pais_sigla, local_nome];
}

function selecao()
{
    const cidadeSelecionada = cidade.value;
    if (cidadeSelecionada) 
    {
        return cidadeSelecionada;
    } 
    else 
    {
        return null;
    }
}

async function buscar_onClick()
{   
    const selecao_cidade = selecao()
    
    if (selecao_cidade !== null)
    {
        try
        {

            const urlCEP = `${BASE_URL}q=${selecao_cidade}&appid=${API_KEY}&units=metric&lang=pt_br`;
            const resposta = await fetch(urlCEP);
            
            if (resposta.ok)
            {   
                const dadosJSON = await resposta.json(); 
                const info = apresentarDados(dadosJSON);

                spanLongitude.innerText = info[0];
                spanLatitude.innerText = info[1];
                spanCandicao.innerText = info[2];
                spanDescricao.innerText = info[3];
                spanTemp.innerText = `${info[4]} °C`;
                spansensT.innerText = `${info[5]} °C`;
                spanTempMax.innerText = `${info[6]} °C`;
                spanTempMin.innerText = `${info[7]} °C`;
                spanUmidade.innerText = `${info[8]}%`;
                spanVelVento.innerText = `${info[9]} m/s`;
                spanPais.innerText = info[10];
                spanLocal.innerText = info[11];
                
                saida.textContent = "Dados encontrados!";
                saida.style.color = "green"

            }
            else
            {
                saida.textContent = "Cidade não encontrada. Verifique o nome e tente novamente.";
                saida.style.color = "red"
            }
        } catch(err) {
            saida.textContent = "Erro ao buscar informações. Verifique sua conexão com a internet.";
            saida.style.color = "red"

        }
    }
    else 
    {
        saida.textContent = "Por favor, selecione uma cidade.";
        saida.style.color = "navy"

    }
}

function limpar_onClick()
{
    spanLongitude.innerText = "-";
    spanLatitude.innerText = "-";
    spanCandicao.innerText = "-";
    spanDescricao.innerText = "-";
    spanTemp.innerText = "-";
    spansensT.innerText = "-";
    spanTempMax.innerText = "-";
    spanTempMin.innerText = "-";
    spanUmidade.innerText = "-";
    spanVelVento.innerText = "-";
    spanPais.innerText = "-";
    spanLocal.innerText = "-";
    
    saida.textContent = "";
}

btnBuscar.addEventListener("click", buscar_onClick);
btnLimpar.addEventListener("click", limpar_onClick);
